### PR TITLE
fixing array parameters to not be copied - only effective change

### PR DESCRIPTION
**Pull request by nivanov-ati**

### CODE REVIEW

File: generator/gen_bindings.go [hunk 1]
```diff
@@ -546,16 +546,8 @@ func (gen *Generator) proxyArgFromGo(memTip tl.Tip, name string,
...
 		return proxy, helper.Nillable
 	case isPlain && goSpec.Slices != 0: // ex: []byte
-		gen.submitHelper(sliceHeader)
-		if goSpec.Base == "unsafe.Pointer" &&
-			(len(goSpec.Raw) == 0 || goSpec.Raw == "unsafe.Pointer") {
-			// Go 1.8+
-			cgoSpec.Base = "unsafe.Pointer"
-		}
-		helper := gen.getCopyBytesHelper(cgoSpec)
-		gen.submitHelper(helper)
```
Comment (jbcpollak) [inline generator/gen_bindings.go]: what is all of this doing? and why do we no longer need it?

File: generator/gen_bindings.go [hunk 2]
```diff
@@ -546,16 +546,8 @@ func (gen *Generator) proxyArgFromGo(memTip tl.Tip, name string,
...
-		proxy = fmt.Sprintf("%s((*sliceHeader)(unsafe.Pointer(&%s)))", helper.Name, name)
-		return proxy, helper.Nillable
```
Comment (jbcpollak) [inline generator/gen_bindings.go:558]: why don't we need to return the proxy and Nillable anymore?

**Comment by jbcpollak** [comment]
@nivanov-ati this just builds out of the box for me

```
jpollak@Yggdrasil:~/src/ati/c-for-go$ go version
go version go1.22.0 linux/amd64
jpollak@Yggdrasil:~/src/ati/c-for-go$ go build
jpollak@Yggdrasil:~/src/ati/c-for-go$ ./c-for-go
Copyright (c) 2015-2017 Maxim Kupriianov <max@kc.vc>
Based on a C99 compiler front end by Jan Mercl <0xjnml@gmail.com>

Usage: ./c-for-go package1.yml [package2.yml] ...
See https://github.com/xlab/c-for-go for examples and documentation.
....
```

File: generator/gen_bindings.go [hunk 3]
```diff
@@ -546,16 +546,8 @@ func (gen *Generator) proxyArgFromGo(memTip tl.Tip, name string,
 		proxy = fmt.Sprintf("%s(%s)", helper.Name, name)
 		return proxy, helper.Nillable
 	case isPlain && goSpec.Slices != 0: // ex: []byte
-		gen.submitHelper(sliceHeader)
```
Comment (jbcpollak) [inline generator/gen_bindings.go]: this looks like its checking if the "sliceHeader" has any dependencies, and if so triggering them to be generated as well, I'm not sure we want to disable that?

**Comment by jbcpollak** [comment]
also, this PR to upstream looks similar with some slight changes:

https://github.com/xlab/c-for-go/pull/160/files

**Comment by nivanov-ati** [comment]
Oh yeah that might be a better way to express this! How did you find it?

**Comment by jbcpollak** [comment]
I just went to check and see if there was any recent activity upstream and just looked at the open PRs

File: generator/gen_bindings.go (hunk 2 above)
Comment (nivanov-ati) [reply generator/gen_bindings.go:558]: you return proxy by default - thats what an empty return statement does. I'm actually not sure what the default return value for it is in our case. In general, I used the diff between uldaq.go of the standard c-for-go translation + my hand edits and the uldaq.go that's generated by this modified c-for-go as a way to diagnose my changes

**Comment by nivanov-ati** [comment]
@jbcpollak re: building - this also builds out of the box for me. However, try debugging main.go in VSCode - for me, it couldnt use "process.go" while it was sharing the package with "main.go"

**Review by jbcpollak — APPROVED** [review]

**Pull request merged**